src/cnxml/serialize.js

```diff
@@ -142,6 +143,30 @@ function list(obj, children) {
 }
 
 
+/**
+ * Serializer for source elements.
+ */
+function source(obj, children) {
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(obj.getText().trim(), 'text/xml')
+    let el = document.createElement('template')
+    el.innerHTML = xmlDoc.documentElement.outerHTML
+
+    const error = xmlDoc.getElementsByTagName('parsererror')
+    if (error.length) {
+        const attrs = {
+            class: 'invalid-xml',
+        }
+        return <div {...attrs}>
+            Error: {error[0].textContent}
+            Content: {obj.getText()}
+        </div>
```

Comment:
If there are any errors with the XML fragment we want them reported immediately, not quietly. Documents with XML errors should not be allowed to save.